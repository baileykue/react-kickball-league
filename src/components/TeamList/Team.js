import React from 'react';
import PlayerLink from '../../components/PlayerList/PlayerLink';

export default function Team({ team }) {
  return (
    <>
      {team.players.map((player) => (
        <PlayerLink key={player.id} player={player} />
      ))}
    </>
  );
}
