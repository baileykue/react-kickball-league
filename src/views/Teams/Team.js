import React, { useState, useEffect } from 'react';
import { getTeamById } from '../../services/teams';
import TeamDetail from '../../components/TeamList/TeamDetail';

export default function Team(props) {
  const id = props.match.params.id;
  const [team, setTeam] = useState([]);
  const [loading, setLoading] = useState(true);
  // const params = useParams();
  //this is a way to get the correct id without passing team.js as a prop in app js

  useEffect(() => {
    const fetchData = async () => {
      const data = await getTeamById(id);
      //const data = await fetchById(params.id);
      setTeam(data[0]);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) return <h3>Loading team...</h3>;

  return (
    <>
      <TeamDetail team={team} showDetail />
    </>
  );
}
