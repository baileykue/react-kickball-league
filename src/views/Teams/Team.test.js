import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Team from './Teams';

test("should render a team's details", async () => {
  const { container } = render(
    <MemoryRouter>
      <Team match={{ params: { id: 4 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Mt. Hoodies');
  expect(container).toMatchSnapshot();
});
