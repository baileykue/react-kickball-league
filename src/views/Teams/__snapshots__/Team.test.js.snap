// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a team's details 1`] = `
<div>
  <div>
    <h1>
      These are the Teams
    </h1>
    <div>
      <a
        href="/teams/1"
      >
        Bridge City Sneakers
      </a>
    </div>
    <div>
      <a
        href="/teams/3"
      >
        Lakeville Thunderfeet
      </a>
    </div>
    <div>
      <a
        href="/teams/4"
      >
        Mt. Hoodies
      </a>
    </div>
    <div>
      <a
        href="/teams/2"
      >
        Stumptown Lumberjacks
      </a>
    </div>
    <div>
      <a
        href="/teams/5"
      >
        Vancougers
      </a>
    </div>
  </div>
</div>
`;
